components/automate-ui/src/app/ui.component.ts

```diff
@@ -61,7 +62,14 @@ export class UIComponent implements OnInit {
   ngOnInit(): void {
     this.router.events.pipe(
         filter(event => event instanceof ActivationStart)
-    ).subscribe((event: any) => this.renderNavbar = !event.snapshot.data.hideNavBar);
+    ).subscribe((event: ActivationStart) => this.renderNavbar = !event.snapshot.data.hideNavBar);
+
+    this.router.events.pipe(
+        filter(event => event instanceof NavigationEnd)
+    ).subscribe(() => this.store.dispatch(new GetAllUserPerms()));
+
+    // Initial call
+    this.store.dispatch(new GetAllUserPerms());
```

Comment:
➡️ Start here:
(1) This loads all introspection data at time of browser refresh.

Reply:
@tarablack01 or @scottopherson should correct me if I'm wrong, but I'd heard one of ui-team's tech debt goals is to remove ui.component.ts (in favor of `app.component.ts`). So might be a good idea to put this in `app.component` now to save some work down the line

Reply:
I checked with them; does not seem to be on the horizon at the moment. But thanks for raising the question.